pkg/secrets/fetch_secret.go

```diff
@@ -49,6 +54,41 @@ func execCommand(inputPayload string) ([]byte, error) {
 	defer cancel()
 
 	cmd := exec.CommandContext(ctx, secretBackendCommand, secretBackendArguments...)
+
+	if len(secretBackendCommandSHA256) != 0 {
```

Comment:
```suggestion
	if secretBackendCommandSHA256 != "" {
```

Reply:
This seems like a personal preference here, mind sharing why you feel this is better style?

Reply:
I think the main logic behind is `len` is more tailored if you plan on iterating over each element of something. While ` != ""` highlight that we're checking for an exact value (empty string here).

It doesn't really matter in the end since the go compiler is going to generate the same code. But I found  ` != ""` clearer since it emphasis that we're working with a string (not a slice or anything else).

it's a detail feel free to ignore.